sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningAwareFileIndex.scala

```diff
@@ -62,6 +62,10 @@ abstract class PartitioningAwareFileIndex(
     pathGlobFilter.forall(_.accept(file.getPath))
   }
 
+  protected lazy val recursiveFileLookup = {
+    parameters.getOrElse("recursiveFileLookup", "false").toBoolean
```

Comment:
shall we document the option in `DataFrameReader`?

Reply:
@Ngone51 Could you submit a follow-up PR to document this? This affects all the built-in file sources. We need to update the documentation of both PySpark and Scala APIs.

Reply:
FYI, there is a Jira about adding this documentation which you will want to reference: [SPARK-29903](https://issues.apache.org/jira/browse/SPARK-29903)

Reply:
@nchammas Could you submit a PR to fix readwriter.py for supporting this new option?

Reply:
Sure, will do. I suppose we'll do that separately from adding the docs, which will get their own PR, correct?

Reply:
Guys, we should also update `DataStreamReadaer` and `streaming.py`.

Reply:
Ok, I'll submit a PR to document it. @gatorsmile

Reply:
FYI @Ngone51: #26718